HPXMLtoOpenStudio/resources/battery.rb

```diff
@@ -56,19 +63,55 @@ def self.apply(model, battery)
     elcs.setBatteryMass(battery_mass)
     elcs.setBatterySurfaceArea(battery_surface_area)
     elcs.setDefaultNominalCellVoltage(default_nominal_cell_voltage)
-    elcs.setCellVoltageatEndofNominalZone(default_nominal_cell_voltage)
     elcs.setFullyChargedCellCapacity(default_cell_capacity)
+    elcs.setCellVoltageatEndofNominalZone(default_nominal_cell_voltage)
+    if not voltage_dependence
+      elcs.setBatteryCellInternalElectricalResistance(0.0002)
+      elcs.setFullyChargedCellVoltage(default_nominal_cell_voltage)
+      elcs.setCellVoltageatEndofExponentialZone(default_nominal_cell_voltage)
+    end
 
-    model.getElectricLoadCenterDistributions.each do |elcd|
-      next unless elcd.inverter.is_initialized
+    elcds = model.getElectricLoadCenterDistributions
+    elcds = elcds.select { |elcd| elcd.inverter.is_initialized } # i.e., not generators
+    if elcds.empty?
+      elcd = OpenStudio::Model::ElectricLoadCenterDistribution.new(model)
+      elcd.setName('Battery elec load center dist')
+      elcd.setElectricalBussType('AlternatingCurrentWithStorage')
+    else
+      elcd = elcds[0] # i.e., pv
 
-      elcd.setElectricalBussType('DirectCurrentWithInverterDCStorage')
-      elcd.setMinimumStorageStateofChargeFraction(minimum_storage_state_of_charge_fraction)
-      elcd.setMaximumStorageStateofChargeFraction(maximum_storage_state_of_charge_fraction)
+      elcd.setElectricalBussType('DirectCurrentWithInverterACStorage')
       elcd.setStorageOperationScheme('TrackFacilityElectricDemandStoreExcessOnSite')
-      elcd.setElectricalStorage(elcs)
-      elcd.setDesignStorageControlDischargePower(rated_power_output)
-      elcd.setDesignStorageControlChargePower(rated_power_output)
+    end
+
+    elcd.setMinimumStorageStateofChargeFraction(minimum_storage_state_of_charge_fraction)
+    elcd.setMaximumStorageStateofChargeFraction(maximum_storage_state_of_charge_fraction)
+    elcd.setElectricalStorage(elcs)
+    elcd.setDesignStorageControlDischargePower(rated_power_output)
+    elcd.setDesignStorageControlChargePower(rated_power_output)
+
+    if not schedules_file.nil?
+      charging_schedule = schedules_file.create_schedule_file(col_name: SchedulesFile::ColumnBatteryCharging)
+      discharging_schedule = schedules_file.create_schedule_file(col_name: SchedulesFile::ColumnBatteryDischarging)
+
+      if (not charging_schedule.nil?) && (not discharging_schedule.nil?)
+        charging_schedule_values = schedules_file.schedules[SchedulesFile::ColumnBatteryCharging]
+        discharging_schedule_values = schedules_file.schedules[SchedulesFile::ColumnBatteryDischarging]
+        values = charging_schedule_values.zip(discharging_schedule_values)
+        if values.any? { |v| v[0] > 0 && v[1] > 0 }
+          fail 'Cannot simultaneously charge and discharge the battery.'
+        end
```

Comment:
I see tests for both error messages introduced here, nice. 👍

We should also add language to the documentation so a user is aware of these restrictions (can't have non-zero values at the same time, both schedules are required if one is provided).

Reply:
On second thought, what is the benefit to having two columns versus just one? If there was a single `battery` column that could go from -1 to 1, then A) this error wouldn't be possible and B) there would be half as much data in the user's CSV file. It seems simpler, but I may be missing a nuance.

Reply:
I guess the benefit to having two schedule columns is the ease in which we can assign the ScheduleFile. If there was just one column, I suppose there'd be a bit of processing, beforehand, to separate it out into charging and discharging schedules.

Reply:
I see. If that's the only benefit, I guess I have a preference for providing the user one less way to screw up their inputs. We already have precedent for having to process schedule values (setpoints F->C), so if it's not that much effort, I think it's worth it.

Reply:
Good point. I'll look into it further.